### PR TITLE
Generate habitatmap_stdized, habitatmap_terr, watersurfaces_hab 2023: a few tweaks

### PR DESCRIPTION
**Pull request by cecileherr**
Main changes (see also https://github.com/inbo/n2khab-preprocessing/pull/65#issuecomment-1996880418)

- generate habitatmap_stdized: update renv , check and correct geometry, remove rbbvos+
- generate habitatmap_terr: update renv 
- generate watersurfaces_hab: update renv , import layers with corrected geometries so that we do not need to correct geometry here (I totally removed the geometry fix since habmap_stdized is already correct and I can import watersurfaces with a correction through `read_watersurfaces(fix_geom = TRUE)` )
- miscellaneous/habitatmap.Rmd: remove the checks (that were for habitatmap_stdized and are now included in generate habiatatmap_stdized) and add checks for habitatmap, incl. a check for invalid geometries. Note: apparently the epsg of the habitatmap is not 31370 but a very similar one (with rounded values) (?)

TO DO:
- I will update n2khab read functions next. Normally only minor changes, the actual read functions are already compatible with the new data sources.
- update the links to Zenodo when the new datasets will be online
- (next version?): in generate habmap_stdized: import habitatmap with corrected geometry (`read_habitatmap(fix_geom = TRUE)`) and remove geometry checks and corrections from generate habitatmap_stdized
- (later): version handling
- miscellaneous/habitatmap.Rmd: look for a solution for the slow st_is_valid step. Or report a bug if confirmed as such

FILES (rendered bookdown files)
[generating_habitatmap_stdized.zip](https://github.com/inbo/n2khab-preprocessing/files/15417972/generating_habitatmap_stdized.zip)
[generating_habitatmap_terr.zip](https://github.com/inbo/n2khab-preprocessing/files/15417978/generating_habitatmap_terr.zip)
[generating_watersurfaces_hab.zip](https://github.com/inbo/n2khab-preprocessing/files/15417980/generating_watersurfaces_hab.zip)
[habitatmap.zip](https://github.com/inbo/n2khab-preprocessing/files/15417995/habitatmap.zip) (with a few minor textual differences with the present code, I will upload the exact version later)

### CODE REVIEW

**Review by florisvdh — COMMENTED** [review]
Apologies for this review coming so late.

### generate_xxx (3 projects)

All of this is great work :ok_hand:. You have added various enhancements and clarifications to these projects!

Also thanks for fixing #68, and for addressing #60 in the considered raw data sources.

I've also compared and verified checksums with the googlesheet and the files that you uploaded, all was fine.

The data sources are ready to be uploaded to Zenodo. I will do this for `habitatmap_terr` (2 versions) after this PR and #65 are merged; @ToonHub can do it for the other two data sources unless we have some urgency to release {n2khab} before he's able to do the uploads.

Remaining notes:

- I have very few and only minor suggestions (no re-run needed of these bookdowns).

- I think it's most elegant git-wise if you add the handling of [this just added PR #65 comment](https://github.com/inbo/n2khab-preprocessing/pull/65#discussion_r1709779323) to the current PR? Those two lines just need to be dropped. Let me know if I better do it.

- Considering your 'to do':

    > (next version?): in generate habmap_stdized: import habitatmap with corrected geometry (read_habitatmap(fix_geom = TRUE)) and remove geometry checks and corrections from generate habitatmap_stdized

    Absolutely. Can you add an issue so that we keep track of this?

    > miscellaneous/habitatmap.Rmd: look for a solution for the slow st_is_valid step. Or report a bug if confirmed as such

    Let's talk about this one!

### habitatmap.Rmd

Thanks for cleaning `habitatmap.Rmd` and extending it! :ok_hand: Indeed the long timing of `sf::st_is_valid()` is something to look at, and hopefully get it solved with or without help of other developers.

Some rather cosmetic remarks are in the review.

Regarding your finding about the clashing habitatmap CRS: you are right! This problem is known and it's the reason why raw data sources are read by {n2khab} with `read_sf(crs = 31370)` or using `st_crs() <-`. The CRS of the raw data source is numerically identical to **`ESRI:102499`** (i.e. only some EPSG identifiers or labels can be different or missing), but not to `EPSG:31370`.

- See https://github.com/inbo/n2khab/issues/85#issuecomment-741604642 for explanation on `ESRI:102499` vs. `EPSG:31370`. Thanks for referring https://github.com/OSGeo/PROJ/issues/1781 which indeed is about the same problem. Even though it appears that ArcMap has resolved their faulty definition, I'm afraid this CRS may or will remain present in official data sources; newer data source versions are often based on older ones and won't necessarily have their CRS reset.
- See https://github.com/inbo/n2khab-preprocessing/pull/50#pullrequestreview-589646581 for advice on implementing this in generate_xxx projects not using {n2khab} functions.
- The solution was set in 2019 already after some comparisons:

https://github.com/inbo/n2khab-preprocessing/blob/3524021ba9d265e5cd425b7ef30d864237996d79/src/miscellaneous/crs_checks.Rmd#L180-L187

You might want to amend `habitatmap.Rmd` given this extra info, although I don't consider this a reason for not merging the PR (you can do that in a new PR as well).

File: src/miscellaneous/habitatmap.Rmd [hunk 1]
```diff
@@ -17,69 +17,272 @@ output:
...
+end_time - start_time
+
+# important: since we need 40 minutes to calculate invalid_geoms, it is probably not 
+# an option to use the same code as in read_watersurfaces
+# the calculation of n_invalid would be too slow, so probably better to drop the if
+# taken from read_watersurfaces: 
+ # if (fix_geom) {
+ #      n_invalid <- sum(
+ #        !st_is_valid(watersurfaces) | is.na(st_is_valid(watersurfaces))
+ #      )
```
[the remark this answers is not in the document]
Comment (cecileherr) [reply src/miscellaneous/habitatmap.Rmd:137]: in https://github.com/inbo/n2khab-preprocessing/pull/70/commits/e8fb23b33e19b762cffcb91aacfa494597e223b6 done for habitatmap

**Comment by cecileherr** [comment]
Follow-up:

> Remaining notes:
> 

> * I think it's most elegant git-wise if you add the handling of [this just added PR #65 comment](https://github.com/inbo/n2khab-preprocessing/pull/65#discussion_r1709779323) to the current PR? Those two lines just need to be dropped. Let me know if I better do it.

=> done in commit https://github.com/inbo/n2khab-preprocessing/pull/70/commits/d0ed2156ce6119c5375c6d4888a0f20f11487e7b

> * Considering your 'to do':
>   > (next version?): in generate habmap_stdized: import habitatmap with corrected geometry (read_habitatmap(fix_geom = TRUE)) and remove geometry checks and corrections from generate habitatmap_stdized
>   Absolutely. Can you add an issue so that we keep track of this?

=> created a new issue https://github.com/inbo/n2khab-preprocessing/issues/72#

>   > miscellaneous/habitatmap.Rmd: look for a solution for the slow st_is_valid step. Or report a bug if confirmed as such
>   
>   
>   Let's talk about this one!

=> after a few trials during the summer 2024: might be partly computer related (problems not reproducible on all computers). Something to keep in mind in the future. 

> Regarding your finding about the clashing habitatmap CRS: you are right! This problem is known and it's the reason why raw data sources are read by {n2khab} with `read_sf(crs = 31370)` or using `st_crs() <-`. The CRS of the raw data source is numerically identical to **`ESRI:102499`** (i.e. only some EPSG identifiers or labels can be different or missing), but not to `EPSG:31370`. (...)
> You might want to amend `habitatmap.Rmd` given this extra info, although I don't consider this a reason for not merging the PR (you can do that in a new PR as well).

=> added this info to miscellaneous/habitatmap.Rmd (commit https://github.com/inbo/n2khab-preprocessing/pull/70/commits/1074ae396c0a4fe5ea5eb4ee2637f4d4e4290994)

An a few remaining suggestions are handled in commit https://github.com/inbo/n2khab-preprocessing/pull/70/commits/e8fb23b33e19b762cffcb91aacfa494597e223b6

**Comment by cecileherr** [comment]
I think I could handle the remaining suggestions, so I will merge this PR. 
(Please let me know if you think I missed anything and I can make the needed changes in a next PR)

**Comment by florisvdh** [comment]
Excellent work @cecileherr! Thanks for the clear [narrative](https://github.com/inbo/n2khab-preprocessing/pull/70#issuecomment-2378706576), it makes it easy to track things.

**Pull request merged**